Update JS Guidlines (#171)
IntelliTect/CodingGuidelines: docs/coding/javascript.md

```
These should all be followed by a semi-colon. In addition the `return` value expression must start on the same line as the return keyword in order to avoid semi-colon insertion.

- :heavy_check_mark: DO use a semi-colon except where it's not allowed

> - :heavy_check_mark: DO configure ESLint rule `semi` with option `["error", "always"]`.

Reviewer: iirc, TypeScript needs to set this rule to `off` to work properly. Since you've added ESLint recommendations, do you think we should always add related TS rules?

``` json
"@typescript-eslint/semi": ["error", "always"],
"semi": "off",
```

Author: I wasn't sure if we wanted to have a dedicated set of guidelines for Typescript or not. I'll add a note about this nuance for now.

## Control Structures
These include `if`, `for`, `while`, `switch`, etc. Here is an example if statement, since it is the most complicated of them:
```js
if (condition1 || condition2) {
  action1();
}
else if (condition3 && condition4) {
  action2();
}
else {
  defaultAction();
}
```

> Control statements should have a new line between the control keyword and opening parenthesis, to distinguish them from function calls.

Reviewer: Thank you for making this change. We'll need to update the description to match, though.

```suggestion
Control statements should have single spaces between the control keyword, control condition, and opening parenthesis.
```
